dev/PushNotifications/PushNotificationReceivedEventArgs.cpp

```diff
@@ -60,51 +49,45 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         return { payload, payload + (length * sizeof(uint8_t)) };
     }
 
-    std::vector<uint8_t> PushNotificationReceivedEventArgs::BuildPayload(winrt::hstring const& payload)
-    {
-        std::string payloadToSimpleString{ ::winrt::Microsoft::Windows::PushNotifications::Helpers::WideStringToUtf8String(payload) };
-        return { payloadToSimpleString.c_str(), payloadToSimpleString.c_str() + (payloadToSimpleString.length() * sizeof(uint8_t)) };
-    }
-
     winrt::com_array<uint8_t> PushNotificationReceivedEventArgs::Payload()
     {
         return { m_rawNotificationPayload.data(), m_rawNotificationPayload.data() + (m_rawNotificationPayload.size() * sizeof(uint8_t)) };
     }
 
     winrt::BackgroundTaskDeferral PushNotificationReceivedEventArgs::GetDeferral()
     {
-        if (!m_unpackagedAppScenario)
+        if (!PushNotificationHelpers::IsPackagedAppScenario())
         {
-            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
-
-            return m_backgroundTaskInstance.GetDeferral();
-        }
-        else
-        {
-            auto dummyDeferral = winrt::make<PushNotificationDummyDeferral>();
-            return dummyDeferral.as<winrt::BackgroundTaskDeferral>();
+            return winrt::make<PushNotificationDummyDeferral>().as<BackgroundTaskDeferral>();
         }
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, m_isForegroundActivation, "Foreground activation cannot call this.");
+
+        return m_backgroundTaskInstance.GetDeferral();
     }
 
     winrt::event_token PushNotificationReceivedEventArgs::Canceled(winrt::BackgroundTaskCanceledEventHandler const& handler)
     {
-        if (!m_unpackagedAppScenario)
+        if (!PushNotificationHelpers::IsPackagedAppScenario())
         {
-            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
-
-            return m_backgroundTaskInstance.Canceled(handler);
+            return winrt::event_token{};
         }
         else
         {
-            return { 0 };
+            THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, m_isForegroundActivation, "Foreground activation cannot call this.");
+
+            return m_backgroundTaskInstance.Canceled(handler);
         }
     }
 
     void PushNotificationReceivedEventArgs::Canceled(winrt::event_token const& token) noexcept
     {
-        if (!m_unpackagedAppScenario)
+        if (!PushNotificationHelpers::IsPackagedAppScenario())
```

Comment:
Nit: simplify and avoid an extra line.

if (PushNotificationHelpers::IsPackagedAppScenario())
{
      THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, m_isForegroundActivation, "Foreground activation cannot call this.");

            m_backgroundTaskInstance.Canceled(token);
}